docs: deploy to aws with env variables and multiple aws profiles (#18045)
gatsbyjs/gatsby: docs/docs/deploying-to-s3-cloudfront.md

That's it!
Run `npm run build && npm run deploy` to do a build and have it immediately deployed to S3!

## Deploying with `.env` variables

Some deployments require passing environment variables. To deploy with environment variables, update the deployment script to your `package.json`:

> ```js

Reviewer: We can add the filename to the code block for an extra tip:
```suggestion
```js:title=package.json
```

"scripts" : {
    ...
    "deploy": "npm run -n \"-r dotenv/config\" && npm run build && gatsby-plugin-s3 deploy"
}
```

> This command requires running `dotenv` first before deploying to s3. Additionally, if you have multiple AWS profiles in your machine, you can deploy by declaring your `AWS_PROFILE` before the deploy script:

Reviewer: It would help to describe what exactly is happening in this script. I also don't see references to dotenv in this doc yet. Do we need to instruct users to install it before they can run this?

Reviewer: Cool! It would be worth mentioning that so that people know. Can you add a note about it?

That's it!
Run `npm run build && npm run deploy` to do a build and have it immediately deployed to S3!

## Deploying with `.env` variables

Some deployments require passing environment variables. To deploy with environment variables, update the deployment script to your `package.json`:

```js:title=package.json
"scripts" : {
    ...
    "deploy": "npm run -n \"-r dotenv/config\" && npm run build && gatsby-plugin-s3 deploy"
}
```

> This command requires `dotenv` first, runs build next, and finally deploys to s3. `dotenv`, already a dependency of Gatsby, loads environment variables and makes it available globally.

Reviewer: ```suggestion
This command requires `dotenv` first, runs build next, and finally deploys to s3. `dotenv`, a dependency of Gatsby that doesn't need to be installed directly, loads environment variables and makes them available globally.
```

Author: thank you for your patience with me! I pushed changes :)

Reviewer: Thank _you_! This looks great!
